packages/react-devtools-shared/src/devtools/views/Components/EditableValue.js

```diff
@@ -67,71 +87,59 @@ export default function EditableValue({
 
       const {key} = event;
 
-      if (key === 'Enter') {
-        if (dataType === 'number') {
-          const parsedValue = parseFloat(editableValue);
-          if (!Number.isNaN(parsedValue)) {
-            overrideValueFn(path, parsedValue);
-          }
-        } else {
-          overrideValueFn(path, editableValue);
+      if (key === 'Enter' && isValid) {
+        const parsedEditableValue = JSON.parse(sanitizeForParse(editableValue));
+
+        if (value !== parsedEditableValue) {
+          overrideValueFn(path, parsedEditableValue);
         }
 
         // Don't reset the pending change flag here.
         // The inspected fiber won't be updated until after the next "inspectElement" message.
         // We'll reset that flag during a subsequent render.
       } else if (key === 'Escape') {
-        setEditableValue(value);
+        setEditableValue(stringifiedValue);
         setHasPendingChanges(false);
+        setIsValid(true);
       }
     },
-    [editableValue, dataType, overrideValueFn, path, value],
+    [editableValue, isValid, dataType, overrideValueFn, path, value],
   );
 
-  // Render different input types based on the dataType
-  let type = 'text';
-  if (dataType === 'boolean') {
-    type = 'checkbox';
-  } else if (dataType === 'number') {
-    type = 'number';
-  }
-
-  let inputValue = value == null ? '' : value;
+  let inputValue = value === undefined ? '' : stringifiedValue;
   if (hasPendingChanges) {
-    inputValue = editableValue == null ? '' : editableValue;
+    inputValue = editableValue;
   }
 
   let placeholder = '';
-  if (value === null) {
-    placeholder = '(null)';
-  } else if (value === undefined) {
+  if (value === undefined) {
     placeholder = '(undefined)';
-  } else if (dataType === 'string') {
-    placeholder = '(string)';
+  } else {
+    placeholder = 'Enter valid JSON';
```

Comment:
I think it would be good to have this placeholder for non `undefined` values. What do you think @bvaughn ?